components/merkledb/src/backends/rocksdb.rs

```diff
@@ -180,38 +234,51 @@ impl Snapshot for RocksDBSnapshot {
         }
     }
 
-    fn iter<'a>(&'a self, name: &str, from: &[u8]) -> Iter<'a> {
-        use rocksdb::{Direction, IteratorMode};
-        let iter = match self.db.cf_handle(name) {
-            Some(cf) => self
-                .snapshot
-                .iterator_cf(cf, IteratorMode::From(from, Direction::Forward))
-                .unwrap(),
-            None => self.snapshot.iterator(IteratorMode::Start),
-        };
-        Box::new(RocksDBIterator {
-            iter: iter.peekable(),
-            key: None,
-            value: None,
-        })
+    fn iter(&self, name: &ResolvedAddress, from: &[u8]) -> Iter<'_> {
+        Box::new(self.rocksdb_iter(name, from))
     }
 }
 
 impl<'a> Iterator for RocksDBIterator<'a> {
     fn next(&mut self) -> Option<(&[u8], &[u8])> {
-        if let Some((key, value)) = self.iter.next() {
-            self.key = Some(key);
-            self.value = Some(value);
-            Some((self.key.as_ref()?.as_ref(), self.value.as_ref()?.as_ref()))
-        } else {
-            None
+        if self.ended {
+            return None;
+        }
+
+        let (key, value) = self.iter.next()?;
+        if let Some(ref prefix) = self.prefix {
+            if &key[..8] != prefix {
+                self.ended = true;
+                return None;
+            }
         }
+
+        self.key = Some(key);
```

Comment:
How would it be initialized in this case? `Option<Box<T>>` [has the same memory layout](https://doc.rust-lang.org/std/option/index.html#options-and-pointers-nullable-pointers) as `Box<T>`. (The `None` variant corresponds to a null pointer in ~plebian~ less memory-safe programming languages.) Thus, I don't think the change would lead to performance improvement, and I'd argue that the current implementation is understandable.

Reply:
Didn't know about layout, then I think current approach is fine.